### DISCUSSION TITLE
Consuming breakpoint by plugin.

### DISCUSSION THREAD

**Discussion by aaron-dadum** [x64dbg/x64dbg#2683] [Ideas]
Hi Guys!

I was wondering, is it possible to consume a breakpoint by tracing plugin? I intercept CB_BREAKPOINT allright, but then the control is taken away from the plugin, and given to the gui, so user has to click on step over or anything similar to pass control back to my plugin so that the trace could continue. 

Is it possible for my plugin to "consume" the BP, so that it never looses control?

Thank you!

**Comment by mrexodia** [comment]
You can use DbgCmdExecDirect(“run”), which should resume execution when
called from the breakpoint callback.

**Reply by aaron-dadum** [reply]
Thank you!
